zebrad/tests/acceptance.rs

```diff
@@ -38,37 +37,47 @@ fn generate_no_args() -> Result<()> {
 #[test]
 fn generate_args() -> Result<()> {
     zebra_test::init();
+    let (mut tempdir, _guard) = tempdir(false)?;
 
     // unexpected free argument `argument`
-    let (child, _guard) = get_child(&["generate", "argument"])?;
+    let child = get_child(&["generate", "argument"], &tempdir)?;
     let output = child.wait_with_output()?;
     output.assert_failure()?;
 
     // unrecognized option `-f`
-    let (child, _guard) = get_child(&["generate", "-f"])?;
+    let child = get_child(&["generate", "-f"], &tempdir)?;
     let output = child.wait_with_output()?;
     output.assert_failure()?;
 
     // missing argument to option `-o`
-    let (child, _guard) = get_child(&["generate", "-o"])?;
+    let child = get_child(&["generate", "-o"], &tempdir)?;
     let output = child.wait_with_output()?;
     output.assert_failure()?;
 
+    // add a config file to path
+    tempdir.push("zebrad.toml");
+
     // Valid
-    let (child, _guard) = get_child(&["generate", "-o", "file.yaml"])?;
+    let child = get_child(
+        &["generate", "-o", tempdir.to_str().unwrap()],
+        &tempdir.parent().unwrap().to_path_buf(),
+    )?;
+
     let output = child.wait_with_output()?;
     output.assert_success()?;
 
-    // Todo: Check if the file was created
+    // Check if the file was created
+    assert!(tempdir.exists());
```

Comment:
Issue: On this line, tempdir isn't a directory any more.

So let's clone it, push the config file name, and use a different variable name for the generated config file.

Reply:
Thanks, a lot better and easier to read. Done at https://github.com/ZcashFoundation/zebra/pull/859/commits/f9cc9bde4d7cefa1fc326c97f315ab766f5a607c